yb-voyager/src/tgtdb/suites/oracleSuite.go

```diff
@@ -0,0 +1,206 @@
+		if err != nil {
+			return "", fmt.Errorf("parsing epoch nanoseconds: %v", err)
+		}
+		epochSeconds := epochNanoSecs / 1000000000
+		epochNanos := epochNanoSecs % 1000000000
+		nanoTimestamp := time.Unix(epochSeconds, epochNanos).Local()
+		oracleTimestampFormat := "02-01-2006 03.04.05.000000000 PM" //format: DD-MON-YYYY HH.MI.SS.FFFFFFFFF PM
+		formattedTimestamp := nanoTimestamp.Format(oracleTimestampFormat)
+		if formatIfRequired {
+			formattedTimestamp = fmt.Sprintf("'%s'", formattedTimestamp)
+		}
+		return formattedTimestamp, nil
+	},
+	"io.debezium.time.ZonedTimestamp": func(columnValue string, formatIfRequired bool) (string, error) {
+		debeziumFormat := "2006-01-02T15:04:05Z07:00"
+
+		parsedTime, err := time.Parse(debeziumFormat, columnValue)
+		if err != nil {
+			return "", fmt.Errorf("parsing timestamp: %v", err)
+		}
+
+		oracleFormat := "06-01-02 3:04:05.000000000 PM -07:00"
+		if parsedTime.Location().String() == "UTC" { //LOCAL TIMEZONE Case
+			oracleFormat = "06-01-02 3:04:05.000000000 PM" // TODO: for timezone ones sqlldr is inserting as GMT though GMT/UTC is similar
+		}
+		formattedTimestamp := parsedTime.Format(oracleFormat)
+		if formatIfRequired {
+			formattedTimestamp = fmt.Sprintf("'%s'", formattedTimestamp)
+		}
+		return formattedTimestamp, nil
+	},
+	"BYTES": func(columnValue string, formatIfRequired bool) (string, error) {
+		//decode base64 string to bytes
+		decodedBytes, err := base64.StdEncoding.DecodeString(columnValue) //e.g.`////wv==` -> `[]byte{0x00, 0x00, 0x00, 0x00}`
+		if err != nil {
+			return columnValue, fmt.Errorf("decoding base64 string: %v", err)
+		}
+		//convert bytes to hex string e.g. `[]byte{0x00, 0x00, 0x00, 0x00}` -> `0000000`
+		hexString := ""
+		for _, b := range decodedBytes {
+			hexString += fmt.Sprintf("%02x", b)
+		}
+		hexValue := hexString
+		if formatIfRequired {
+			hexValue = fmt.Sprintf("'%s'", hexString) // in insert statement no need of escaping the backslash and add quotes
+		}
+		return string(hexValue), nil
+	},
+	"MAP": func(columnValue string, _ bool) (string, error) {
+		mapValue := make(map[string]interface{})
+		err := json.Unmarshal([]byte(columnValue), &mapValue)
+		if err != nil {
+			return columnValue, fmt.Errorf("parsing map: %v", err)
+		}
+		var transformedMapValue string
+		for key, value := range mapValue {
+			transformedMapValue = transformedMapValue + fmt.Sprintf("\"%s\"=>\"%s\",", key, value)
+		}
+		return fmt.Sprintf("'%s'", transformedMapValue[:len(transformedMapValue)-1]), nil //remove last comma and add quotes
+	},
+	"STRING": func(columnValue string, formatIfRequired bool) (string, error) {
+		if formatIfRequired {
+			formattedColumnValue := strings.Replace(columnValue, "'", "''", -1)
+			return fmt.Sprintf("'%s'", formattedColumnValue), nil
+		} else {
+			return columnValue, nil
+		}
+	},
+	"INTERVAL YEAR TO MONTH": func(columnValue string, formatIfRequired bool) (string, error) {
```

Comment:
uff, the interval handling looks like a lot of manually written logic 😅  
Have you checked interval.handling.mode in debezium? THere is a numeric option. Maybe that might be ingestible to oracle directly?

Reply:
I can check that. Currently, the way debezium dumps interval data is easily ingested using COPY as is to YB but with SQlldr we need some transformation.

Reply:
Ah, right. forgot that we also need to worry about YB 😅 
Ok, just see if the numeric option makes it easy. Otherwise, if you've tested this logic, leave it for now, and we can revisit as this is not a very highly used data type?

Reply:
I checked the numeric type it gives values in microseconds e.g. -
```
"1","-44706600000000","-1560000000"
```
 COPY can't work with this. I haven't tested sqlldr yet but I think it will also not be able to handle this and need to be manipulated to some understandable format. 
I think the current string format is fine. I have also tested my logic and it works fine.